spec/inertia/middleware_spec.rb

```diff
@@ -28,6 +28,24 @@
 
       it { is_expected.to eq 303 }
     end
+
+    it 'is thread safe' do
+      delete_request_proc = -> { delete redirect_test_path, headers: { 'X-Inertia' => true } }
+      get_request_proc = -> { get empty_test_path }
+
+      statusses = []
+
+      threads = []
+
+      100.times do
+        threads << Thread.new { statusses << delete_request_proc.call }
+        threads << Thread.new { get_request_proc.call }
+      end
+
+      threads.each(&:join)
+
+      expect(statusses.uniq).to eq([303])
```

Comment:
I ran this test with the original buggy code, and it will occasionally pass by random luck. I'm not worried about that, since the code structurally prevents the issue really well and I would not expect to see a regression here, but figure it's worth pointing out in case it ever pops up as confusing in the future